samples/canopen/canopen_rpdo/RPDOCanNode.hpp

```diff
@@ -0,0 +1,226 @@
+/**
+ * Representation of the CAN node. Handles constructing the object
+ * dictionary and other baseline settings. The idea is that each "board"
+ * will have a specific object dictionary associated with it. The object
+ * dictionary itself will also need to have information on "data of interest".
+ * For example, a temperature management system may to expose water pump
+ * flow rate in the object dictionary.
+ */
+#include <stdint.h>
+
+#include <co_core.h>
+
+class RPDOCanNode {
+public:
+    RPDOCanNode();
+
+    /**
+     * Expose a way to programmatically update the sampleData.
+     *
+     * An example use case could be temperature data that is collected
+     * then stored in this object. Under the hood, the object dictionary
+     * has the address of the temperature data, so the triggered PDO
+     * would also effectively be updated.
+     *
+     * @param newValue[in] The value to set sample data to
+     */
+    void setSampleDataA(uint8_t newValue);
+    void setSampleDataB(uint16_t newValue);
+
+    /**
+     * Get the contained sample data
+     *
+     * @return The value of the sample data
+     */
+    uint8_t getSampleDataA();
+    uint16_t getSampleDataB();
+
+    /**
+     * Get a pointer to the start of the object dictionary
+     *
+     * @return Pointer to the start of the object dictionary
+     */
+    CO_OBJ_T* getObjectDictionary();
+
+    /**
+     * Get the number of elements in the object dictionary.
+     *
+     * @return The number of elements in the object dictionary
+     */
+    uint8_t getNumElements();
+
+    /**
+     * The node ID used to identify the device on the CAN network.
+     */
+    static constexpr uint8_t NODE_ID = 0x02;
```

Comment:
Because of issues we've had with communication in the past, all node IDs should be expressed in decimal, not hex